python-lib/plugin_partitioning_utils.py

```diff
@@ -0,0 +1,144 @@
+
+    # Check partitioning
+    if not folder.get_definition().get("partitioning"):
+        return ""
+
+    # Get partitioning file path pattern
+    file_path_pattern = folder.get_definition().get("partitioning").get("filePathPattern")
+
+    # Get discrete partitioning dimensions
+    dimensions = folder.get_definition().get("partitioning").get("dimensions")
+    discrete_dimensions = [dimension.get("name") for dimension in dimensions if dimension.get("type") != "time"]
+
+    # Generate path placeholders replacements dictionnary
+    path_replacement_dict = DEFAULT_OUTPUT_PATH_REPLACEMENT_DICT.copy()
+    for dimension in discrete_dimensions:
+        replacement_key = f"%{{{dimension}}}"
+        # Get flow variable containing dimension's value
+        replacement_value = dataiku.get_flow_variables()["DKU_DST_" + dimension]
+        path_replacement_dict[replacement_key] = replacement_value
+
+    # Generate path
+    dst_file_path = file_path_pattern
+    for dimension_placeholder in path_replacement_dict.keys():
+        dst_file_path = dst_file_path.replace(dimension_placeholder, path_replacement_dict[dimension_placeholder])
+
+    dst_file_path = "".join(dst_file_path.rsplit(".*"))
+
+    return dst_file_path
+
+
+@time_logging(log_message="Getting input partitions paths")
+def get_input_partitions_paths(folder_id: AnyStr) -> List[AnyStr]:
+    """Function returning the paths to the input folder's required partitions.
+    If input folder isn't partitioned, returns an empty string
+    Args:
+        folder_id: input folder's id
+    Returns:
+        List of strings containing the input partitions' paths
+    """
+    # Create folder handle
+    client = dataiku.api_client()
+    folder = client.get_project(dataiku.default_project_key()).get_managed_folder(folder_id)
+
+    # Check partitioning
+    if not folder.get_definition().get("partitioning"):
+        return [""]
+
+    # Instanciate result
+    src_file_paths = []
+
+    # Get partitioning file path pattern
+    file_path_pattern = folder.get_definition().get("partitioning").get("filePathPattern")
+
+    # Get partitioning dimensions
+    dimensions = folder.get_definition().get("partitioning").get("dimensions")
+
+    # Get time dimension name if defined
+    try:
+        time_dimension = [dimension.get("name") for dimension in dimensions if dimension.get("type") == "time"][0]
+    except IndexError:
+        time_dimension = None
+
+    # Get discrete partitioning dimensions
+    discrete_dimensions = [dimension.get("name") for dimension in dimensions if dimension.get("type") != "time"]
+
+    # Get all necessary values per partitioning dimension
+    dimension_names = [dimension.get("name") for dimension in dimensions]
+    values_per_dimension = {
+        dimension_name: dataiku.get_flow_variables()[f"DKU_SRC_{dimension_name}_VALUES"].replace("'", "").split(", ")
+        for dimension_name in dimension_names
+    }
```

Comment:
maybe you can fusion the three loop to do only one pass over dimensions and avoid using `except IndexError`.
and also use a function to process each dimension